src/components/SignUp/SignUp.js

```diff
@@ -54,6 +84,76 @@ export default class SignUp extends Component {
     }
   };
 
+  validateName = e => {
```

Comment:
improvements to consider : you can move your validation functions to separate class and reuse them across the application